### PR TITLE
(#1527) Merge Threads and Timed together and use RepeatedTest

### PR DESCRIPTION
**Pull request by victornoel**
For #1527 + a bit of cleanup

### CODE REVIEW

File: src/test/java/org/cactoos/experimental/ThreadsTest.java [hunk 1]
```diff
@@ -25,65 +25,103 @@
...
+    @RepeatedTest(ThreadsTest.REPETITIONS)
+    void containsResults() throws Exception {
+        final ExecutorService extor = Executors.newFixedThreadPool(3);
+        try {
+            new Assertion<>(
+                "Must contain results from callables",
+                new Threads<String>(
+                    extor,
+                    Duration.ofSeconds(1),
+                    () -> {
```
Comment (baudoliver7) [inline src/test/java/org/cactoos/experimental/ThreadsTest.java:68]: @victornoel How about declaring three callables property of `ThreadsTest` (`firsttext`, `secondtext`, `thirdtext`) and initializing them in setUp method ? I notice that it is repeated in each test.

File: src/test/java/org/cactoos/experimental/ThreadsTest.java [hunk 2]
```diff
@@ -25,65 +25,103 @@
...
+                        this.sleep();
+                        return "txt 3";
                     }
-                }
-            }
-        );
+                ),
+                new HasValues<>(
+                    "txt 1",
```
Comment (baudoliver7) [inline src/test/java/org/cactoos/experimental/ThreadsTest.java:82]: @victornoel we could avoid to repeat these three values each time (txt 1, txt 2 and txt 3) by declaring an iterable variable of these values.

File: src/test/java/org/cactoos/experimental/ThreadsTest.java [hunk 3]
```diff
@@ -92,37 +130,106 @@ void containsResults() {
...
+     *  {@link ExecutorService} was initiated by {@link Threads} itself.
+     */
+    @RepeatedTest(ThreadsTest.REPETITIONS)
+    void containsValuesWithInlineExecutorService() throws Exception {
         new Assertion<>(
-            "wraps error into CompletionException",
-            () -> new Threads<String>(
-                Executors.newSingleThreadExecutor(),
-                new ScalarOf<>(
+            "Must contain results from the callables",
```
Comment (baudoliver7) [inline src/test/java/org/cactoos/experimental/ThreadsTest.java:170]: @victornoel perhaps we could write : "Must contain results from the callables when using inline executor service" to make a difference with test `containsResults`

File: src/test/java/org/cactoos/experimental/ThreadsTest.java [hunk 4]
```diff
@@ -92,37 +130,106 @@ void containsResults() {
...
-        this.repeat(
-            arg -> new Assertion<>(
-                // @checkstyle LineLength (1 line)
-                "contains results from the callables when using the inline executor service",
+    @RepeatedTest(ThreadsTest.REPETITIONS)
+    void containsResultsNoTimeout() throws Exception {
+        final ExecutorService extor = Executors.newFixedThreadPool(3);
+        try {
+            new Assertion<>(
+                "Must contain results from callables",
```
Comment (baudoliver7) [inline src/test/java/org/cactoos/experimental/ThreadsTest.java:230]: @victornoel perhaps we could write : "Must contain results from the callables without using timeout" to make a difference with test `containsResults`

File: src/test/java/org/cactoos/experimental/ThreadsTest.java [hunk 5]
```diff
@@ -137,24 +244,72 @@ void containsValuesWithInlineExecutorService() {
...
-    private void repeat(final Proc<?> test) {
-        new UncheckedFunc<>(
-            new Repeated<>(
-                arg -> {
-                    test.exec(null);
-                    return null;
+    @RepeatedTest(ThreadsTest.REPETITIONS)
+    void containsValuesWithInlineExecutorServiceNoTimeout() {
+        new Assertion<>(
+            "Must contain results from the callables",
```
Comment (baudoliver7) [inline src/test/java/org/cactoos/experimental/ThreadsTest.java:295]: @victornoel perhaps we could write : "Must contain results from the callables when using inline executor without timeout" to make a difference with test `containsResults`

**Review by baudoliver7 — CHANGES_REQUESTED** [review]
@victornoel Please, see comments below.

File: src/test/java/org/cactoos/experimental/ThreadsTest.java (hunk 1 above)
Comment (victornoel) [reply src/test/java/org/cactoos/experimental/ThreadsTest.java:68]: @baudoliver7 it  is a good practice not to share knowledge between tests, it makes them more independent and more resilent, see https://www.yegor256.com/2016/05/03/test-methods-must-share-nothing.html for a related argument :)

**Comment by victornoel** [comment]
@baudoliver7 thx, see the changes and my comment

File: src/test/java/org/cactoos/experimental/ThreadsTest.java (hunk 1 above)
Comment (baudoliver7) [reply src/test/java/org/cactoos/experimental/ThreadsTest.java:68]: @victornoel Good !

**Review by baudoliver7 — APPROVED** [review]
@victornoel thx, it is ok for me :)

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @baudoliver7[/z](https://www.0crat.com/u/baudoliver7), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**